src/transformers/modeling_flax_utils.py

```diff
@@ -708,7 +713,21 @@ def from_pretrained(
                             "proxies": proxies,
                             "use_auth_token": use_auth_token,
                         }
-                        if has_file(pretrained_model_name_or_path, WEIGHTS_NAME, **has_file_kwargs):
+                        # check if an index file exists
+                        if has_file(pretrained_model_name_or_path, WEIGHTS_INDEX_NAME, **has_file_kwargs):
+                            if from_pt:
+                                resolved_archive_file = cached_file(
+                                    pretrained_model_name_or_path, WEIGHTS_INDEX_NAME, **cached_file_kwargs
+                                )
+                            if resolved_archive_file is not None:
+                                is_sharded = True
```

Comment:
This should all go above, just after the part from flax sharded checkpoints. This part should only raise errors with a message to tell the user to use `from_pt=True`.